### PR TITLE
feat: support recursive field conversion for RECORD type columns

### PR DESCRIPTION
**Pull request by katamotokosuke**
## Summary
- Add recursive field type conversion for RECORD type columns based on `fields` definition (TIMESTAMP, DATETIME, DATE, TIME, BOOLEAN, etc.)
- Support REPEATED mode and nested RECORD fields
- Add pre-release workflow to automatically publish gems with commit hash suffix to GPR on pull requests

## Changes
- `value_converter_factory.rb`: Add `fields` parameter and implement recursive type conversion via `build_record_field_converters` / `apply_field_converters` methods
- `test_value_converter_factory.rb`: Add comprehensive tests covering all field types, REPEATED mode, nested RECORD, and edge cases (+219 lines)
- `README.md`: Expand documentation for `fields` option
- `.github/workflows/pull-request.yml`: Add new pre-release workflow for PR builds
- `.github/workflows/gem-push.yml`: Update to actions/checkout@v4 and push-gem-to-gpr-action@v2

### Basic RECORD with type conversion
```yaml
column_options:
  - name: data
    type: RECORD
    fields:
      - {name: key, type: STRING}
      - {name: ts, type: TIMESTAMP, timestamp_format: "%Y-%m-%dT%H:%M:%S"}
```
Input: `{"key": "hello", "ts": "2024-01-15T10:30:00"}`
Output: `{"key": "hello", "ts": "2024-01-15 10:30:00.000000 +00:00"}`

### Nested RECORD with timezone
```yaml
column_options:
  - name: data
    type: RECORD
    fields:
      - name: inner
        type: RECORD
        fields:
          - {name: ts, type: TIMESTAMP, timestamp_format: "%Y-%m-%dT%H:%M:%S", timezone: "Asia/Tokyo"}
          - {name: label, type: STRING}
```
Input: `{"inner": {"ts": "2024-01-15T10:30:00", "label": "value"}}`
Output: `{"inner": {"ts": "2024-01-15 10:30:00.000000 +09:00", "label": "value"}}`

### REPEATED RECORD
```yaml
column_options:
  - name: data
    type: RECORD
    fields:
      - name: items
        type: RECORD
        mode: REPEATED
        fields:
          - {name: ts, type: TIMESTAMP, timestamp_format: "%Y-%m-%dT%H:%M:%S"}
          - {name: name, type: STRING}
```
Input: `{"items": [{"ts": "2024-01-15T10:30:00", "name": "item1"}, {"ts": "2024-02-20T15:00:00", "name": "item2"}]}`
Output: `{"items": [{"ts": "2024-01-15 10:30:00.000000 +00:00", "name": "item1"}, {"ts": "2024-02-20 15:00:00.000000 +00:00", "name": "item2"}]}`

### Various field types
```yaml
column_options:
  - name: data
    type: RECORD
    fields:
      - {name: str, type: STRING}
      - {name: int, type: INTEGER}
      - {name: flt, type: FLOAT}
      - {name: bool, type: BOOLEAN}
      - {name: ts, type: TIMESTAMP, timestamp_format: "%Y-%m-%dT%H:%M:%S"}
      - {name: dt, type: DATETIME, timestamp_format: "%Y/%m/%d %H:%M:%S"}
      - {name: d, type: DATE}
      - {name: t, type: TIME}
```

## Test plan
- [ ] Unit tests pass
- [ ] Pre-release workflow runs on PR
- [ ] Pre-release gem is published to GPR
- [ ] Verify BigQuery transfer works correctly with pre-release gem in dev environment

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/embulk/output/bigquery/value_converter_factory.rb [hunk 1]
```diff
@@ -322,6 +329,54 @@ def json_converter
             raise NotSupportedType, "cannot take column type #{type} for json column"
           end
         end
+
+        def build_record_field_converters
```
Comment (isab) [inline lib/embulk/output/bigquery/value_converter_factory.rb:333]: Do we need to include the `strict` setting for the nested records or does it apply to the nested level as well? 

ネストレコードに「strict」設定を含める必要がありますか、それともネストされたレベルにも適用されますか？

**Review by isab — COMMENTED** [review]
一つ質問がある 🙏

File: lib/embulk/output/bigquery/value_converter_factory.rb (hunk 1 above)
Comment (katamotokosuke) [reply lib/embulk/output/bigquery/value_converter_factory.rb:333]: https://github.com/trocco-io/embulk-output-bigquery/pull/47/commits/1b55c9b25e6cde032edde0e9518e9393fe0a108c

**Review by isab — APPROVED** [review]
